src/codegen/sys/lib_.rs

```diff
@@ -389,20 +440,67 @@ fn generate_classes_structs(w: &mut Write, env: &Env, classes: &[&Class]) -> Res
                 w,
                 "{comment}#[repr(C)]\n{comment}pub struct {name}(c_void);\n",
                 comment = comment,
-                name = klass.c_type
+                name = klass.c_type,
+            ));
+            try!(writeln!(
+                w,
+                "impl ::std::fmt::Debug for {name} {{\n\
+                    \tfn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {{\n\
+                        \t\twrite!(f, \"{name} @ {{:?}}\", self as *const _)\n\
+                    \t}}\n\
+                }}\n",
+                name = klass.c_type,
             ));
         } else {
+            let can_generate_fields_debug = can_generate_fields_debug(&klass.fields);
             try!(writeln!(
                 w,
-                "{comment}#[repr(C)]\n{comment}pub struct {name} {{",
+                "{comment}#[repr(C)]\n{debug}{comment}pub struct {name} {{",
```

Comment:
This line seems generated struct with extra line:
```
#[repr(C)]

pub struct AtkAttribute {
    pub name: *mut c_char,
    pub value: *mut c_char,
}
```

Reply:
Wo. Where does this backline comes from? Oo

Reply:
Yes very strange